pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1976,7 +1976,7 @@ func (ws *WorkflowStatus) IsTaskResultIncomplete(name string) bool {
 	if found {
 		return !value
 	}
-	return true
+	return false
```

Comment:
could you add a comment here to clarify this behavior? from an out-of-context reading, this would be unexpected / unintuitive
```suggestion
	return false // workflows from older versions do not have this status, so assume completed if this is missing
```